source/margarita/upgrading.rst

```diff
@@ -0,0 +1,557 @@
+
+Updating Production
+-------------------
+
+If staging updates successfully, it's time to take care of the production machine. Follow the steps
+above, starting from the :ref:`Update Repo Code <update-repo>` pathway above, but on the production
+machine.
+
+If you get this far and everything is working then it's time to celebrate!! Make sure that the
+*develop* and *master* branches are properly updated with the changes in *your-feature-branch* and
+that ``repo.branch`` is set to the correct value in ``conf/pillar/<environment>.sls``.
+
+Troubleshooting
+---------------
+
+Here are some issues that may or may not come up during your upgrade. As we find new issues, we
+should update the docs above (if they are general), or add them here, if they aren't general or
+we're not sure.
+
+* ``newrelic_license_key`` must be capitalized. Some projects have a secret for
+  ``newrelic_license_key``, but the current margarita uses ``NEW_RELIC_LICENSE_KEY``
+
+* NewRelic settings may need adjusting, which you can do via environment variables (see
+  other documentation in this repo for details).
+
+* If you get timeout errors during the first deploy, it may be because of a few different issues.
+
+  * Low CPU/RAM servers might need the salt timeouts extended. Add ``timeout: 600`` to
+    ``/etc/salt/master`` and ``/etc/salt/minion`` (or edit the value if already present) and then
+    restart both the ``salt-master`` and ``salt-minion``. Wait a **full minute** or so before
+    starting any salt command. My salt-minions took a loooong time to start on a low-powered box.
+
+  * There might be a chicken/egg problem with the firewall. Do a grep for ``UFW BLOCK`` in
+    ``/var/log/syslog``::
+
+      Jan 13 21:19:48 ip-<deleted> kernel: [78350448.038946] [UFW BLOCK] IN=eth0 OUT= MAC=<deleted> SRC=<deleted> DST=<deleted> LEN=60 TOS=0x00 PREC=0x00 TTL=45 ID=33808 DF PROTO=TCP SPT=53084 DPT=4506 WINDOW=14600 RES=0x00 SYN URGP=0
+
+    That says port 4506 (DPT) is being blocked by the firewall. If so, run::
+
+      # ufw allow salt
+
+  If salt is installed, the command above will open up the ports that salt needs. Our deploy does
+  that too, but if the firewall is already running, then our salt state can't run. I'm still
+  confused how this problem happened on a server which had already been running salt successfully,
+  but ¯\\_(ツ)_/¯.
+
+* *VAGRANT NOTE*: Make sure to undo the ``Vagrantfile`` setting which syncs the conf folder to
+  ``/srv`` on the VM, because the project no longer expects that folder to be synced, so will run
+  into problems trying to change permissions on the files there.
+
+  Change this::
+
+    config.vm.synced_folder "conf/", "/srv/"
+
+  to::
+
+    config.vm.synced_folder "conf/", "/srv/", disabled: true
+
+  and then restart the VM. If you are using the new ``Vagrantfile``, you shouldn't need to do that.
+
+* *VAGRANT NOTE:* Remove the ``source`` and ``public`` symlinks as we rsync now rather than symlink.
```

Comment:
Consistency nit: here the ``:`` is included in the italicization, but in the previous bullet it isn't.